etuutt_bot/utils/role.py

```diff
@@ -15,16 +15,17 @@ async def parse_roles(file: PurePath | str) -> dict[str, list[str]]:
 
     ues = {}
     for category in raw_ues.split(";"):
-        if category != "" and category.split(":")[0] in parse_categories().keys():
+        if category != "" and category.split(":")[0] in parse_categories():
             ues[category.split(":")[0]] = category.split(":")[1].split(",")
 
     return ues
 
 
 @lru_cache(maxsize=None)
 def parse_categories() -> dict[str, int]:
-    if getenv("UES_CATEGORIES") is not None:
-        categories = {}
-        for category in getenv("UES_CATEGORIES").split(","):
-            categories[category.split(":")[0].upper()] = int(category.split(":")[1])
-        return categories
+    categories = getenv("UES_CATEGORIES")
+    if categories is None:
+        raise KeyError("Environment variable missing : UES_CATEGORIES")
+    return {
+        key.upper(): int(val) for cat in categories.split(",") for (key, val) in cat.split(":")
```

Comment:
```suggestion
        key.upper(): int(val) for cat in categories.split(",") for (key, val) in [cat.split(":")]
```
Oubli de compréhension de liste ?

Reply:
Je pense qu'un KeyError, c'est approprié, étant donné que techniquement, il manque une clef dans les paires clef-valeur des variables d'environnement. C'est l'erreur qui est retournée, en tout cas, quand on tente de faire un `os.environ["CLEF_INEXISTANTE"]`. La gestion est explicite juste pour le message d'erreur.

Un ValueError, c'est quand c'est la valeur de la variable qui est mauvaise. Là, c'est la clef le problème.

Et OsError, jsp.

Reply:
Ok ça me va pour KeyError, par contre faut l'exception n'est pas atteinte si la variable est présente mais vide dans le .env.
Ce qui nous donne ça comme erreur :
[ERROR   ] bot: Command 'addall' raised an exception: ValueError: not enough values to unpack (expected 2, got 1).